onion-routing-htlc-forwarding.asciidoc

```diff
@@ -19,3 +19,48 @@ Relevant questions to answer:
+If there are 20 hops in the payment, then all 20 layers of the packet contain information about the twenty routing nodes.
+
+Let us now consider the adverse case, where the packet size is NOT fixed i.e. every time a layer is peeled off of the packet, the size of the packet reduces.
+If a malicious routing node receives a packet, it can use the size of the packet to estimate how many "layers" are left in this packet.
+If it receives a packet and estimates that there are 20 layers left, i.e. a full packet, then it knows that the node who sent it the packet is the originator of the payment.
+If it receives a packet and estimates that there is 1 layer left, then it knows that the node that it is sending the packet to is the final recipient of the payment.
+Even if it receives a packet and estimates that there are 2 layers left, then it knows that it is either transmitting the packet to the last routing node (i.e. the payment recipient) or it is transmitting the packet to the second last routing node before the recipient.
+It can graph out all the channels of the second last routing node and it knows that the final recipient of the payment is either the second last routing node itself, or one of their channel partners (we ignore the case of private channels).
+In all cases, some privacy of the payer and the recipient are lost.
+
+[[malicious-routing-diagram]]
+.If the Malicious node knows there are 2 layers left, then it knows that the payment recipient is either Node 19 (and there were only 19 hops) or one of Node 19's channel partners
+image:images/malicious-routing-diagram.png["If the Malicious node knows there are 2 layers left, then it knows that the payment recipient is either Node 19 (and there were only 19 hops) or one Node 19's channel partners"]
+
+We can extend this example.
+Imagine a malicious entity sets up multiple Lightning nodes connected to other well-connected nodes, and it also connects itself across popular payment routes.
+These malicious Lightning nodes would then be popular routing choices for those wanting to send payments, especially if they set their routing fees low.
+The malicious nodes can then capture the data of all packets that pass through their routing nodes.
+With additional information, such as the names of the other routing nodes, it could infer who is making these payments, who is receiving them, and for what amounts.
+footnote:[Note that not all Lightning nodes are anonymous.
+It is known, for example, that the nodes "aantonop" and "1.ln.aantonop.com" are owned by the author of this book Andreas Antonopolous.
+Furthermore, many companies and businesses in this space use a publicly known name for their routing nodes.
+If we see, for example, a payment with destination "Bitrefill", we could infer that someone is making a purchase from Bitrefill.
```

Comment:
actually we can't as names are just set by the node operator and not assigned by the domain name system. thus anyone can name their node as they wish and we can't really deduce anything from the  `alias`

Reply:
Thanks for the review Rene, I've worded it better in the commit below to include node pubkeys. You're right, I could spoof you myself by naming my node "renepickhardt". But you could claim ownership of your own node by posting on Twitter: "I own a node with X alias and Y pubkey", in which case your node now becomes "known"

https://github.com/lnbook/lnbook/pull/308/commits/ce45e40d2439898067d819b07a39d6f71ce81375

Reply:
agree and I think ok for merging. In general we try to describe the protocol. so what people do outside of it is kind of their problem. of course in general people act outside of the protocol. --> ACK